app/Helpers/database/user-activity.php

```diff
@@ -350,7 +351,7 @@ function getUserGameActivity(string $username, int $gameID): array
 
                     return;
                 }
-                $possibleSession = $session;
+                $possibleSession = &$session;
```

Comment:
This was the original problem. Not grabbing the child by reference didn't propagate the changes back to the caller.